eclipse-sw360/sw360
# feat(ui): add note filed in license page

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: 　#1997

### Suggest Reviewer
> You can suggest reviewers here with an @mention.

# Add note filed in license page - Overview of improvements

## Purpose

Add "note" field in license page, and enter/display comments for each license.

## The outline of changes

- The change of data class\
  Add member "note" to license.
- The changes of frontend(GUI)
  - License additional page： Add "Note" field and a text box to License Details.\
    (Add to the License edit page in the same way, and display current values in the text box as initial values.) !![new](https://github.com/eclipse-sw360/sw360/assets/39894802/c9bd36b3-6481-4157-a190-b348093be314)
  - License detail page：\
    Add "Note" field to License Details, and display value of the text box. ![License_note_details]![image](https://github.com/eclipse-sw360/sw360/assets/39894802/20dc3141-530b-4c74-9364-8ce8350c03cc)
    


    
- The changes of backend
  - Add note accessor\
    Add setter and getter methods to SW360License class.
  - Existing license update process\
    Add note in LicenseDatabaseHandler's license update process in the same way as existing field items.
- The change of RestAPI\
  No change.

## How to test

- GUI
  - License detail page
    1. Create license.
    2. Open detail page of the created license.
    3. Confirm that "Note" field has been added to the License Details.
  - License additional page
    1. Open license additional page.
    2. Confirm that "Note" field and a text box have been added to the License Details.
    3. After entering a comment in the text box and adding a new License, confirm that the comment entered in Note is displayed in detail page.
  - License edit page
    1. Select an existing license with Note, and open the license edit page.
    2. Confirm that the input value is displayed as initial value in the Note text box.
    3. After editing the Note comment and updating the license, confirm that the updated comment in Note is displayed in detail page.

## Limitations

Nothing special to mention.
